File: lib/artwork.rb
class Artwork
  include DataMapper::Resource

  property :id, Serial
  property :name, String
  property :email, String, :format => :email_address
  property :created_at, DateTime
  property :view_count, Integer
  property :vote_count, Integer

  def url
    "/artwork/#{self.id}"
  end

  def to_json(*a)
    {
      'guid' => self.url,
      'name' => self.name,
      'email' => self.email,
      'created_at' => self.created_at,
      'view_count' => self.view_count,
      'vote_count' => self.vote_count
    }.to_json(*a)
  end

  def self.parse_json(body)
    json = JSON.parse(body)

    {
      :guid => json['guid'],
      :name => json['name'],
      :email => json['email'],
      :created_at => json['created_at'],
      :view_count => json['view_count'],
      :vote_count => json['vote_count']
    }
  end
end

File: apps/baba_gallery_web/models/artwork.js
// ==========================================================================
// Project:   BabaGalleryWeb.Artwork
// Copyright: ©2009 My Company, Inc.
// ==========================================================================
/*globals BabaGalleryWeb */

/** @class

  (Document your Model here)

  @extends SC.Record
  @version 0.1
*/
BabaGalleryWeb.Artwork = SC.Record.extend(
/** @scope BabaGalleryWeb.Artwork.prototype */ {

  name: SC.Record.attr(String),
  email: SC.Record.attr(String),
  created_at: SC.Record.attr(SC.DateTime),
  updated_at: SC.Record.attr(SC.DateTime),
  view_count: SC.Record.attr(Number, { defaultValue: 0 }),
  vote_count: SC.Record.attr(Number, { defaultValue: 0 }),
  image_url: SC.Record.attr(String),

  icon: function() {
    return "sc-icon-user-16";
  }.property().cacheable(),

  url: function() {
    return this.get('guid');
  }.property('guid').cacheable()

}) ;
